src/r1cs/sparse.rs

```diff
@@ -204,25 +212,46 @@ mod tests {
 
     let result = sparse_matrix.multiply_vec(&vector);
 
-    // Assuming the multiplication is done row by row,
-    // Result[0] = 2.0 * 2.0 = 4.0
-    // Result[1] = 3.0 * 3.0 = 9.0
-    // Result[2] = 4.0 * 1.0 = 4.0
-    assert_eq!(result, vec![Fr::from(4), Fr::from(9), Fr::from(4)]);
+    assert_eq!(result, vec![Fr::from(25), Fr::from(9), Fr::from(4)]);
   }
 
-  #[test]
-  fn test_matrix_iter() {
-    type Fr = <G as Group>::Scalar;
-    let matrix_data = vec![
-      (0, 1, Fr::from(2)),
-      (1, 2, Fr::from(3)),
-      (2, 0, Fr::from(4)),
-    ];
-    let sparse_matrix = SparseMatrix::<Fr>::new(&matrix_data, 3, 3);
+  // For working around the orphan trait impl rule
+  /// Wrapper struct around a field element that implements additional traits
+  #[derive(Clone, Debug, PartialEq, Eq)]
+  pub struct FWrap<F: PrimeField>(pub F);
+
+  impl<F: PrimeField> Copy for FWrap<F> {}
+
+  #[cfg(not(target_arch = "wasm32"))]
+  /// Trait implementation for generating `FWrap<F>` instances with proptest
+  impl<F: PrimeField> Arbitrary for FWrap<F> {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+      let strategy = any::<[u8; 32]>()
+        .prop_map(|seed| FWrap(F::random(StdRng::from_seed(seed))))
+        .no_shrink();
+      strategy.boxed()
+    }
+  }
```

Comment:
Oh, `FWrap` indeed only exists in Lurk. I think it's ok to push this in a utility module for now. Perhaps `src/r1cs/util.rs`?